ipyparallel/client/asyncresult.py

```diff
@@ -134,6 +135,33 @@ def _init_futures(self):
         self._output_future.add_done_callback(self._resolve_output)
         self.add_done_callback(self._finalize_result)
 
+    def iopub_streaming_output_callback(self, eid, msg):
+        msg_type = msg['header']['msg_type']
+        if msg_type == 'stream':
+            msg_content = msg['content']
+            stream_name = msg_content['name']
+            if stream_name == 'stdout':
+                self._display_stream(msg_content['text'], '[stdout:%i] ' % eid)
+            elif stream_name == 'stderr':
+                self._display_stream(
+                    msg_content['text'], '[stderr:%i] ' % eid, file=sys.stderr
+                )
+
+    @contextmanager
+    def stream_output(self):
+
+        from functools import partial
+
+        for eid, msg_id in zip(self._targets, self.msg_ids):
+            callback_func = partial(self.iopub_streaming_output_callback, eid)
+            self._client._futures[msg_id].iopub_callbacks.append(callback_func)
+
+        yield
+
+        # This throws a KeyError. It seems to be already cleaned up before we get here.
```

Comment:
Get a handle on the future(s) before yielding:

```python
f = self._client._futures[msg_id]
```

Then it will still be available after the yield

Reply:
That worked. Thanks!